### DISCUSSION TITLE
Is it possible to set the SQS consumer queue to LocalStack?

### DISCUSSION THREAD

**Discussion by debnon** [bbc/sqs-consumer#361] [Help] [answered]
I would like to be able to run integration tests using SQS consumer and a LocalStack SQS queue like so
```
const app = Consumer.create({
	queueUrl: "http://localhost:4566/000000000000/sqs-consumer-data",
	handleMessage: async (message) => {
		logger.info(message);
	},
});
```

but I will get an error like
`SQS receive message failed: The address https://sqs.eu-west-2.amazonaws.com/ is not valid for this endpoint.` when trying to use actual credentials
or
`SQS receive message failed: The security token included in the request is invalid.` when trying to use mock credentials that are the same as set up in the LocalStack container

**Comment by nicholasgriffintn** [comment, marked as answer]
It is yes!

You'll need to configure an SQS Client as well, like so:

https://github.com/bbc/sqs-consumer/blob/main/test/features/utils/consumer/handleMessage.js

Which imports this from the file `../sqs`:

```
const sqsConfig = {
  region: 'eu-west-1',
  endpoint: 'http://localhost:4566',
  credentials: {
    accessKeyId: 'key',
    secretAccessKey: 'secret'
  }
};

exports.sqs = new SQSClient(sqsConfig);
```

**Reply by debnon** [reply]
Perfect, thanks!
